src/server/game/Handlers/ItemHandler.cpp

```diff
@@ -702,6 +702,10 @@ void WorldSession::SendListInventory(ObjectGuid vendorGuid)
     // Resize vector to real size (some items can be skipped due to checks)
     packet.Items.resize(count);
 
+    // Repair-only NPC
+    if (!count && vendor->HasNpcFlag(UNIT_NPC_FLAG_REPAIR))
+        packet.Reason = 1;
```

Comment:
I'm not sure if this applies only for repair vendors. Guild Vendors should also show an empty window when you're not in a guild or if you do not have unlocked guild rewards yet.

Reply:
```
ServerToClient: SMSG_VENDOR_INVENTORY (0x25B4) Length: 17 ConnIdx: 1 Time: 09/09/2021 15:46:17.141 Number: 1896
VendorGUID: Full: 0x20425C00003249C00000610000380FB6 Creature/0 R4247/S97 Map: 0 (Reinos del Este) Entry: 51495 (Steeg Haskell) Low: 3674038
Reason: 1
VendorItems: 0
```

You are right. I guess then it will be enough to check that count is 0.

Reply:
So now it's the time where we might need a reverse engineering guy who can suggest names for that reason value as magic numbers are bad.

Reply:
So yeah. Apparently this field is being used for all cases where a vendor window is needed while no items are available. So the check for  GOSSIP_OPTION_VENDOR can be removed entirely imo.

Reply:
https://git.io/Juzp3 that's my take on that

Reply:
Are there no other possible values for this field?

Reply:
Perfect. Then I will use your same constexpr definition and update the pull request

Reply:
Done. Thanks for your help 😄